libbpf-tools/tcptop.bpf.c

```diff
@@ -0,0 +1,215 @@
+    
+    bpf_ringbuf_submit(e, 0);
+}
+
+static __always_inline
+int tcp_sendmsg_handler(struct pt_regs *ctx, struct sock *sk,
+    struct msghdr *msg, size_t size)
+{
+    u32 pid = bpf_get_current_pid_tgid() >> 32;
+
+
+	if (filter_pid && pid != filter_pid) {
+		return 0;
+	}
+
+    //u16 dport = 0, family;
+    u16 family;
+	BPF_CORE_READ_INTO(&family, sk, __sk_common.skc_family);
+
+    if (family == AF_INET) {
+        struct ipv4_key_t ipv4_key = {.pid = pid};
+		BPF_CORE_READ_INTO(&ipv4_key.saddr, sk, __sk_common.skc_rcv_saddr);
+		BPF_CORE_READ_INTO(&ipv4_key.daddr, sk, __sk_common.skc_daddr);
+		BPF_CORE_READ_INTO(&ipv4_key.lport, sk, __sk_common.skc_num);
+        BPF_CORE_READ_INTO(&ipv4_key.dport, sk, __sk_common.skc_dport);
+    	bpf_map_update_elem(&ipv4_send_bytes, &ipv4_key, &size, 0 /* flags: BPF_ANY */);
+        trace_v4(ctx, ipv4_key, true, family);
+
+    } else if (family == AF_INET6) {
+        struct ipv6_key_t ipv6_key = {.pid = pid};
+        BPF_CORE_READ_INTO(&ipv6_key.saddr, sk, __sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
+        BPF_CORE_READ_INTO(&ipv6_key.daddr, sk, __sk_common.skc_v6_daddr.in6_u.u6_addr32);
+        BPF_CORE_READ_INTO(&ipv6_key.lport, sk, __sk_common.skc_num);
+        BPF_CORE_READ_INTO(&ipv6_key.dport, sk, __sk_common.skc_dport);
+        bpf_map_update_elem(&ipv6_send_bytes, &ipv6_key, &size, 0 /* flags: BPF_ANY */);
+        trace_v6(ctx, ipv6_key, true, family);
+    }
+    /*else { //drop | this is commented for testing purposes
+		return 0;
+	}*/
+	return 0;
+}
+
+static __always_inline
+int tcp_cleanup_rbuf_handler(struct pt_regs *ctx, struct sock *sk, int copied)
+{
+    u32 pid = bpf_get_current_pid_tgid() >> 32;
+
+	if (filter_pid && pid != filter_pid) {
+		return 0;
+	}
+
+    u16 family;
+	BPF_CORE_READ_INTO(&family, sk, __sk_common.skc_family);
+
+    if (copied <= 0)
+        return 0;
+
+    if (family == AF_INET) {
+        struct ipv4_key_t ipv4_key = {.pid = pid};
+		BPF_CORE_READ_INTO(&ipv4_key.saddr, sk, __sk_common.skc_rcv_saddr);
+		BPF_CORE_READ_INTO(&ipv4_key.daddr, sk, __sk_common.skc_daddr);
+		BPF_CORE_READ_INTO(&ipv4_key.lport, sk, __sk_common.skc_num);
+        BPF_CORE_READ_INTO(&ipv4_key.dport, sk, __sk_common.skc_dport);
+    	bpf_map_update_elem(&ipv4_recv_bytes, &ipv4_key, &copied, 0 /* flags: BPF_ANY */);
+        trace_v4(ctx, ipv4_key, false, family);
+
+    } else if (family == AF_INET6) {
+        struct ipv6_key_t ipv6_key = {.pid = pid};
+        BPF_CORE_READ_INTO(&ipv6_key.saddr, sk, __sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
+        BPF_CORE_READ_INTO(&ipv6_key.daddr, sk, __sk_common.skc_v6_daddr.in6_u.u6_addr32);
+        BPF_CORE_READ_INTO(&ipv6_key.lport, sk, __sk_common.skc_num);
+        BPF_CORE_READ_INTO(&ipv6_key.dport, sk, __sk_common.skc_dport);
+        bpf_map_update_elem(&ipv6_recv_bytes, &ipv6_key, &copied, 0 /* flags: BPF_ANY */);
+        trace_v6(ctx, ipv6_key, false, family);
+    }
+    /*else { //drop
+		return 0;
+	}*/	
```

Comment:
debug leftovers

Reply:
This was left behind on purpose, just like tcptop.py left it behind. Nevertheless, followed your suggestion and deleted the comment.

Reply:
tcptop.py has "// else drop" comment, it doesn't have commented out code. so feel free to leave a single comment, but not a commented out block of code. Please also prefer C89 style comments (/* */), that's what BPF code in libbpf-tools follow, as it's the kernel style.

Reply:
Changed to "// else drop" as suggested. I understand your point, having some code commented is different than what tcptop.py had. I appreciate your suggestions because I improve as a contributor and as a software engineer.

Reply:
so few nits about that comment:
  1) // is not C89 style comment (which libbpf-tools are trying to follow, in line with what kernel style dictates)
  2) the comment itself is misleading. There is no packet drop, we just ignore other protocols, so something like "/* ignore other protocols */" is way more clear, IMHO.

But I'd just drop the comment altogether, tbh. It's not very useful regardless.